upload/library/Sidane/Threadmarks/EditHistoryHandler/Threadmark.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+class Sidane_Threadmarks_EditHistoryHandler_Threadmark extends XenForo_EditHistoryHandler_Abstract
+{
+    protected $_prefix = 'posts';
+
+    protected function _getContent($contentId, array $viewingUser)
+    {
+      /* @var $postModel XenForo_Model_Post */
+      $postModel = XenForo_Model::create('XenForo_Model_Post');
+      $post = $postModel->getPostById($contentId, array(
+        'join' => XenForo_Model_Post::FETCH_FORUM | XenForo_Model_Post::FETCH_THREAD | XenForo_Model_Post::FETCH_USER |  Sidane_Threadmarks_Model_Post::FETCH_THREADMARKS,
+        'permissionCombinationId' => $viewingUser['permission_combination_id']
+      ));
+      if ($post)
+      {
+        $post['permissions'] = XenForo_Permission::unserializePermissions($post['node_permission_cache']);
+        $post['label'] = $post['threadmark_label'];
+        $post['edit_count'] = $post['threadmark_edit_count'];
+        $post['user_id'] = $post['threadmark_user_id'];
+        unset($post['threadmark_label']);
+        unset($post['threadmark_edit_count']);
+        unset($post['threadmark_user_id']);
+      }      
+      return $post;
+    }
+
+    protected function _canViewHistoryAndContent(array $content, array $viewingUser)
+    {
+      $threadmarkModel = XenForo_Model::create('Sidane_Threadmarks_Model_Threadmarks');
+      return $threadmarkModel->canViewThreadmark($content, $null, $content['permissions'], $viewingUser);
```

Comment:
I wouldn't expect the average forum user to care about the edit history. I would restrict it to users who can edit threadmarks.